### PR TITLE
refactor(client): Extract change password account related logic from the view into models.

### PR DESCRIPTION
**Pull request by shane-tomlinson**
@zaach - r?

A small continuation of the work you started with signIn/signUp.

The change password account management logic was inside of the view, which smelled of mixed concerns. Moved logic to user & account models.

issue #2376

### CODE REVIEW

File: app/tests/spec/models/account.js [hunk 1]
```diff
@@ -614,5 +614,64 @@ function (chai, sinon, p, Constants, Assertion, ProfileClient,
       });
     });
 
+    describe('changePassword', function () {
+      it('returns `INCORRECT_PASSWORD` error if old password is incorrect (event if passwords are the same)', function () {
```
Comment (pdehaan) [inline app/tests/spec/models/account.js]: "event"?

File: app/scripts/models/user.js [hunk 2]
```diff
@@ -224,6 +224,9 @@ define([
 
     signInAccount: function (account, relier) {
       var self = this;
+      // Try to sign the user in before checking whether the
+      // passwords are the same. If the user typed the incorrect old
+      // password, they should know that first.
```
Comment (zaach) [inline app/scripts/models/user.js]: This comment seems to be specific to a password-change flow.

File: app/tests/spec/views/change_password.js [hunk 3]
```diff
@@ -216,87 +209,11 @@ function (chai, $, sinon, AuthErrors, FxaClient, Metrics, p,
...
-                  newPassword,
-                  relier,
-                  {
-                    reason: view.fxaClient.SIGNIN_REASON.PASSWORD_CHANGE,
-                    sessionTokenContext: account.get('sessionTokenContext')
-                  }
-              ));
-              assert.isTrue(user.setSignedInAccount.calledWith(account));
+              assert.isTrue(user.changeAccountPassword.calledWith(
+                  account, oldPassword, newPassword));
```
Comment (zaach) [inline app/tests/spec/views/change_password.js]: Ahh, refreshing to see all those stubs of tangential concerns removed.

**Comment by shane-tomlinson** [comment]
@zaach - updated to remove the inappropriate comment!

**Pull request merged**